app/controllers/user/orders_controller.rb

```diff
@@ -16,7 +16,8 @@ def create
         order.order_items.create({
           item: item,
           quantity: cart.count_of(item.id),
-          price: item.price
+          # price: item.price
+          price: cart.discount_price(item.id)
```

Comment:
If a discount is not active, `.discount_price(item.id)` returns the regular price